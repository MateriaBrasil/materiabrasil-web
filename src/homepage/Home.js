import React, { Fragment } from 'react'
import Sections from './Sections'
import handleScroll from './handleScroll'
import NavBar from '../navBar/NavBar'

export default class extends React.Component {
  constructor(props) {
    super(props)
    this.state = { activeStep: 0 }
    this.handleScroll = handleScroll.bind(this)
  }

  render() {
    return (
      <Fragment>
        <NavBar />
        <Sections handleScroll={this.handleScroll} />
      </Fragment>
    )
  }
}
