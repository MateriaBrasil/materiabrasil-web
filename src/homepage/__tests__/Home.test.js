import React from 'react'
import renderer from 'react-test-renderer'

import Home from '../Home'

jest.mock('react-router-dom/Link', () => props => (
  <div {...props}>Link - {props.children}</div>
))

jest.mock('react-media', () => props => (
  <div {...props}>Media - {props.children}</div>
))

jest.mock('../../navBar/NavBar', () => props => <div {...props}>NavBar</div>)

const createMockMediaMatcher = matches => () => ({
  matches,
  addListener: () => {},
  removeListener: () => {},
})

jest.mock('../Explore', () => props => <div {...props}>Explore</div>)
jest.mock('../HowToUseIt', () => props => <div {...props}>HowToUseIt</div>)
jest.mock('../Introduction', () => props => <div {...props}>Introduction</div>)
jest.mock('../Section', () => props => <div {...props}>{props.children}</div>)

let originalMatchMedia
beforeAll(() => {
  originalMatchMedia = window.matchMedia
})
afterAll(() => {
  window.matchMedia = originalMatchMedia
})

describe('when it matches', () => {
  it('renders correctly', () => {
    window.matchMedia = createMockMediaMatcher(true)
    const props = {}
    const tree = renderer.create(<Home {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('when it does not match', () => {
  it('renders correctly', () => {
    window.matchMedia = createMockMediaMatcher(false)
    const props = {}
    const tree = renderer.create(<Home {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
