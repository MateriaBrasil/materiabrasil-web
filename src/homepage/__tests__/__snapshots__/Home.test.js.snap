// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when it does not match renders correctly 1`] = `
Array [
  <div>
    NavBar
</div>,
  <div
    handleScroll={[Function]}
>
    <div>
        Introduction
    </div>
</div>,
  <div
    handleScroll={[Function]}
>
    <div>
        HowToUseIt
    </div>
</div>,
  <div
    handleScroll={[Function]}
>
    <div>
        Explore
    </div>
</div>,
]
`;

exports[`when it matches renders correctly 1`] = `
Array [
  <div>
    NavBar
</div>,
  <div
    handleScroll={[Function]}
>
    <div>
        Introduction
    </div>
</div>,
  <div
    handleScroll={[Function]}
>
    <div>
        HowToUseIt
    </div>
</div>,
  <div
    handleScroll={[Function]}
>
    <div>
        Explore
    </div>
</div>,
]
`;
